src/CounterInterceptor/WorkerCounter.cs

```diff
@@ -0,0 +1,91 @@
+namespace TemporalioSamples.CounterInterceptor;
+
+using System.Numerics;
+
+public static class WorkerCounter
+{
+    public const string NumberOfWorkflowExecutions = "numOfWorkflowExec";
+    public const string NumberOfChildWorkflowExecutions = "numOfChildWorkflowExec";
+    public const string NumberOfActivityExecutions = "numOfActivityExec";
+    public const string NumberOfSignals = "numOfSignals";
+    public const string NumberOfQueries = "numOfQueries";
+
+    private static Dictionary<string, Dictionary<string, BigInteger?>> perWorkflowIdDictionary =
+        new Dictionary<string, Dictionary<string, BigInteger?>>();
+
+    public static void Add(string workflowId, string type)
+    {
+        if (!perWorkflowIdDictionary.TryGetValue(workflowId, out Dictionary<string, BigInteger?>? value))
+        {
+            value = DefaultInfoMap();
+            perWorkflowIdDictionary.Add(workflowId, value);
+        }
+
+        if (value[type] == null)
```

Comment:
See previous comment, for setting value[type] to 1 or current + 1 with ternary.

Reply:
null check is no longer needed after refactoring to use unit